src/zenml/zen_server/utils.py

```diff
@@ -196,6 +200,36 @@ def initialize_workload_manager() -> None:
             _workload_manager = workload_manager_class()
 
 
+def run_template_executor() -> "BoundedThreadPoolExecutor":
+    """Return the initialized run template executor.
+
+    Raises:
+        RuntimeError: If the run template executor is not initialized.
+
+    Returns:
+        The run template executor.
+    """
+    global _run_template_executor
+    if _run_template_executor is None:
+        raise RuntimeError("Run template executor not initialized")
+
+    return _run_template_executor
+
+
+def initialize_run_template_executor() -> None:
```

Comment:
What about a de-initialize/cleanup function?

Reply:
Added this in a the fastapi shutdown event